valkyrie/lib/valkyrie/storage_adapter.rb

```diff
@@ -44,6 +44,30 @@ class File < Dry::Struct
       def stream
         io
       end
+
+      # @param id [Valkyre::ID]
+      # @param digests [Array<Digest>]
+      # @return [Array<Digest>]
+      def checksum(digests:)
+        io.rewind
+        while (chunk = io.read(4096))
+          digests.each { |digest| digest.update(chunk) }
+        end
+
+        digests.map(&:to_s)
+      end
+
+      # @param id [Valkyre::ID]
+      # @param size [Integer]
+      # @param digests [Array<Digest>]
+      # @return [Boolean]
+      def valid?(size:, digests:)
+        return false if size && io.size.to_i != size.to_i
```

Comment:
Question about this: does this mean if I don't want to test for size I do `valid?(size: false, digests: my_digests)`?

Reply:
Maybe it should be `size: nil` instead, so you'd just do `valid?(digest: my_digests)` instead?

Reply:
:+1:

Reply:
I've updated the `valid?` parameters to make `size` optional, and I've updated the StorageAdapter shared spec to have a file block in the middle (https://github.com/samvera-labs/valkyrie/pull/219/files#diff-63d86542dbc084f293fd44283abb9cbeR26) — I've done this to take advantage of the file we're already uploading, and avoiding having to create another tempfile to upload.